include/settings/settings.h

```diff
@@ -93,6 +149,19 @@ struct settings_handler {
 	 */
 };
 
+/**
+ * Register a static handler for settings items
+ *
+ * @param _handler Structure containing registration info, this must be const
+ *	  The handler name in ROM needs to be unique, it is generated from
+ *	  _handler and the linenumber of SETTINGS_REGISTER_STATIC()
+ *
+ */
+#define SETTINGS_REGISTER_STATIC(_handler)	\
```

Comment:
There are quite many similar "registration" concepts in the tree. It'd be good to use consistent naming throughout the project, and I don't think anything is currently using "register" as the verb, rather "define": `LOG_BACKEND_DEFINE` `BT_GATT_SERVICE_DEFINE` `K_THREAD_DEFINE` (and the `BT_SETTINGS_DEFINE` which you're removing in the subsequent patch). Most kernel objects also have `*_DEFINE()` macros, although they're not exactly registering anything (beyond kernel object debugging which will be able to iterate these objects later if necessary). So I'm wondering is there a good reason to deviate from this naming convention for settings handlers?

Reply:
Agreed, I will change this to SETTINGS_STATIC_HANDLER_DEFINE(), I am unsure if I should change `settings_register` to `settings_dynamic_handler_define()`.

Reply:
Ive also suggested to just pass the handlers parameters and then do the handler declaration within the macro, like most DEFINE macro do, since most of the time the code using it is not doing anything with the variable.

Reply:
I tried to keep it more inline with the `settings_register()` routine. Most defines were also from the pre `Z_STRUCT_SECTION_FOREACH()` definition, with this it seems more clear to define the struct and than to define using the struct. It also allows the removal of any NULL declaration.
If you would rather like: `SETTINGS_STATIC_HANDLER_DEFINE(bt_ccc, "bt/ccc", NULL, bt_ccc_set, NULL, NULL)` I can change it, no problem.

However I cannot stringify "bt/ccc" so it will need to be added, as well as bt_ccc.

Reply:
I have to say there is another advantage to using `SETTINGS_STATIC_HANDLER_DEFINE(bt_ccc, "bt/ccc", ...)`: shippable failed on my PR because I defined `const struct settings_handler_static bt_handler={...`, `bt_handler` was already defined. This would not have been the case when I would have used your suggestion.

Reply:
Reworked to `SETTINGS_STATIC_HANDLER_DEFINE(hname, name, get, set, commit, export)` because `native_posix` didn't seem to like the `const struct settings_handler_static` method.